### DISCUSSION TITLE
Separate actor communication

### DISCUSSION THREAD

**Discussion by jgrant95** [statelyai/xstate#4765] [Q&A]
Hi, fellow noob trying to use xState v5 here 👋 

Simply put I have a multiplayer game that I want to create using state machines.

I have created the following machines; lobby, setup and core-game. These represent what I think our the 3 phases to a game session.

My question is should I be creating a top-level GameSession machine to encapsulate all three elements, or should I treat them as completely separate machines and communicate via subscribing to status events (e.g. `done`). 

My main concern with a top-level machine, is how do I communicate events to my core-game machine? Do I just forward them on from my top-level?

Basically what would be best practice in this scenario please 😄 

All help is appreciated!

**Comment by davidkpiano** [comment]
Yes, your actor structure should be hierarchical. You can use [actor registration](https://stately.ai/docs/system#actor-registration) to register actors system-wide so you can send events to them from any other actor in the system.

**Reply by jgrant95** [reply]
Okay - happy with that, however in that scenario, I'm struggling to know the best way to send events to the child actors.

So if I make a top-level machine that registers many child actors, do I just add an always: sendTo()?
(Which is the v5 alternative to autoForward). My main concern was that it literally says in the migration to v5 docs, it's not generally recommended to do this...?
